### PR TITLE
[ADD][16.0][MIG] account_move_substate

### PR DESCRIPTION
**Pull request by bosd**
standard addition / migration :smiley:

### CODE REVIEW

**Comment by bosd** [comment]
@CasVissers-360ERP Can you please review?

**Review by CasVissers-360ERP — APPROVED** [review]
Functional review.

**Comment by Dranyel-Bosd** [comment]
[Screencast from 2023-10-06 14-34-18.webm](https://github.com/OCA/account-invoicing/assets/147027997/aff79a3f-f654-4fb4-876b-21f40a0018a8)

**Review by Dranyel-Bosd — APPROVED** [review]
[Screencast from 2023-10-06 14-34-18.webm](https://github.com/OCA/account-invoicing/assets/147027997/2d4cffb4-fec9-4f38-a776-22820ec01969)

Looks so good tome :+1: :sparkler:

File: account_move_substate/models/account_move.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+        ondelete={"account.move": "cascade"},
+    )
+
+
+class AccountMove(models.Model):
+    _inherit = ["account.move", "base.substate.mixin"]
+    _name = "account.move"
+    _state_field = "state"
+
+    def _track_template(self, changes):
```
Comment (Saran440) [inline account_move_substate/models/account_move.py:21]: This function will add on base_substate. Can you delete this please
https://github.com/OCA/server-ux/pull/736

Comment (CasVissers-360ERP) [reply account_move_substate/models/account_move.py:21]: @bosd can you take a look here?

Comment (bosd) [reply account_move_substate/models/account_move.py:21]: Will do, Let's first get https://github.com/OCA/server-ux/pull/736 merged. As this one depends on it.

Comment (bosd) [reply account_move_substate/models/account_move.py:21]: @Saran440 Can you re-review? 🙏 
Please make it non blocking. As your proposed refactor is not included yet for V16.

There is still ongoing discussion on the refactor. It should not block this PR.
We can always refactor this module, once the proposed changes are in.

**Comment by bosd** [comment]
@etobella Can you please merge this one? 🙏

**Comment by etobella** [comment]
We shall wait for the merge on server-ux, isn't it?

**Comment by bosd** [comment]
> We shall wait for the merge on server-ux, isn't it?

@etobella  We could wait, but IMO better to merge it as is. 
This PR follows the same structure as all other _substate modules.
Example: [sale_substate](https://github.com/OCA/sale-workflow/tree/16.0/sale_substate)  for V16 which has been merged.

If/when the refactor of the `base_substate` module happens,
We should refactor all the _substate modules. (This one, `sale_substate`, purchase_substate and so on)

Waiting.....  is also blocking backports from this pr. 😉

File: account_move_substate/models/account_move.py (hunk 1 above)
Comment (bosd) [reply account_move_substate/models/account_move.py:21]: Just noticed this is non blocking

**Comment by bosd** [comment]
@Saran440 Do you concur to merge this one as is. And deal with refactoring later when/if it happens?

**Comment by Saran440** [comment]
@bosd I see your commit from v14, edit template and migrate to v15. but v15 is don't have PR?

Should we waiting v14 merge first? https://github.com/OCA/account-invoicing/pull/1001
next, we will migrate from v14 to v16 (if v15 no one used)

**Comment by bosd** [comment]
> but v15 is don't have PR?

I am at Odoo XP. Can open the pr for v15 when I have a bit of free time. 😉

Because it can take some time to get a merge. I opened the migration commits as well. 

If no one uses or reviews v15, we just let it go stale and closed. (As sadly happens with more pr's)

**Comment by bosd** [comment]
Oops, had a long day.
Forgot I made the pr yesterday see https://github.com/OCA/account-invoicing/pull/1595 :grinning:

**Review by Dranyel-Bosd — APPROVED** [review]
Looks so good to me :+1:

**Comment by bosd** [comment]
> We shall wait for the merge on server-ux, isn't it?

@etobella I think we've waited long enough 😄 
Can you 🙏 merge? We can always deal with any refactor later.

**Review by etobella — APPROVED** [review]
I do agree with you @bosd 

We can proceed with merge, because the other module is just blocking it but seems that there will not be any news soon....

/ocabot merge nobump

**Pull request merged**